sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -5716,10 +5707,9 @@ template <atomic_op Op, typename T, int N, typename Toffset,
           typename AccessorTy,
           typename PropertyListT =
               ext::oneapi::experimental::detail::empty_properties_t>
-__ESIMD_API __ESIMD_API std::enable_if_t<
-    __ESIMD_DNS::get_num_args<Op>() == 0 && std::is_integral_v<Toffset> &&
```

Comment:
Is removing the `is_integral_v` check international? I see it in a few places and just want to make sure.

Reply:
I think, this change here is good. It is better to let this function be matched and be lowered to (acc-au0-1) where the check already exists: https://github.com/intel/llvm/blob/6a74b2399a3cc5a3cfab874f9615445e0d29fb27/sycl/include/sycl/ext/intel/esimd/memory.hpp#L5731

Reply:
I mean `static_assert` with a meaningful user-friendly error is better, than error "could not match" with a long list of prototypes that could not be matched.

Reply:
It's removed in many places, so lets at least make sure every call tree has the static assert somewhere